lib/ansible/modules/network/eos/eos_user.py

```diff
@@ -340,6 +341,12 @@ def main():
                            supports_check_mode=True)
 
     warnings = list()
+    if module.params['password'] and not module.params['configured_password']:
```

Comment:
`required_if` seems like the wrong way to go about this, as it isn't an error, it's just a notice that if one has specified password, they probably have the wrong key (unless they have also specified configured_password)